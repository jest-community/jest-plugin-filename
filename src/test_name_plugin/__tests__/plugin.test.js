import { KEYS } from 'jest-watcher';
import pluginTester from '../../test_utils/pluginTester';
import TestNamePlugin from '../plugin';

const testResults = [
  {
    testResults: [{ title: 'foo 1' }, { title: 'bar 1' }],
  },
  {
    testResults: [{ title: 'foo 2' }, { title: 'bar 2' }],
  },
];

it('shows the correct initial state', async () => {
  const { stdout, updateConfigAndRun, plugin, type } = pluginTester(
    TestNamePlugin,
  );

  const runPromise = plugin.run({}, updateConfigAndRun);
  expect(stdout.write.mock.calls.join('\n')).toMatchSnapshot();
  type(KEYS.ENTER);

  await runPromise;
});

it('shows the correct message when there are no cached tests', async () => {
  const { stdout, updateConfigAndRun, plugin, type } = pluginTester(
    TestNamePlugin,
  );

  const runPromise = plugin.run({}, updateConfigAndRun);
  expect(stdout.write.mock.calls.join('\n')).toMatchSnapshot();
  type('t', KEYS.ENTER);

  await runPromise;
});

it('can use arrows to select a specific test', async () => {
  const {
    stdout,
    hookEmitter,
    updateConfigAndRun,
    plugin,
    type,
  } = pluginTester(TestNamePlugin);

  hookEmitter.onTestRunComplete({ testResults });
  const runPromise = plugin.run({}, updateConfigAndRun);
  stdout.write.mockReset();
  type('f', KEYS.ARROW_DOWN, KEYS.ARROW_DOWN, KEYS.ENTER);
  expect(stdout.write.mock.calls.join('\n')).toMatchSnapshot();

  await runPromise;

  expect(updateConfigAndRun).toHaveBeenCalledWith({
    mode: 'watch',
    testNamePattern: 'foo 2',
  });
});

it('can select a pattern that matches multiple tests', async () => {
  const {
    stdout,
    hookEmitter,
    updateConfigAndRun,
    plugin,
    type,
  } = pluginTester(TestNamePlugin);

  hookEmitter.onTestRunComplete({ testResults });
  const runPromise = plugin.run({}, updateConfigAndRun);
  stdout.write.mockReset();
  type('f', 'o', KEYS.ENTER);
  expect(stdout.write.mock.calls.join('\n')).toMatchSnapshot();

  await runPromise;

  expect(updateConfigAndRun).toHaveBeenCalledWith({
    mode: 'watch',
    testNamePattern: 'fo',
  });
});

it('can configure the key and prompt', async () => {
  const { plugin } = pluginTester(TestNamePlugin, {
    key: 'l',
    prompt: 'have a custom prompt',
  });

  expect(plugin.getUsageInfo()).toEqual({
    key: 'l',
    prompt: 'have a custom prompt',
  });
});

it('test matching is case insensitive', async () => {
  const {
    stdout,
    hookEmitter,
    updateConfigAndRun,
    plugin,
    type,
  } = pluginTester(TestNamePlugin);

  hookEmitter.onTestRunComplete({ testResults });
  const runPromise = plugin.run({}, updateConfigAndRun);
  type('f');
  stdout.write.mockReset();
  type('O');
  expect(stdout.write.mock.calls.join('\n')).toMatchSnapshot();
  type(KEYS.ENTER);
  await runPromise;
});
