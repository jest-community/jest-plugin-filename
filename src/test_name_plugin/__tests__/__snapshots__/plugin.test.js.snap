// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can select a pattern that matches multiple tests 1`] = `



 pattern › f
[MOCK - cursorSavePosition]


 Pattern matches 2 tests from cached test suites

 › foo 1

 › foo 2
[MOCK - cursorTo(12, 5)]
[MOCK - cursorRestorePosition]



 pattern › fo
[MOCK - cursorSavePosition]


 Pattern matches 2 tests from cached test suites

 › foo 1

 › foo 2
[MOCK - cursorTo(13, 5)]
[MOCK - cursorRestorePosition]
`;

exports[`can use arrows to select a specific test 1`] = `



 pattern › f
[MOCK - cursorSavePosition]


 Pattern matches 2 tests from cached test suites

 › foo 1

 › foo 2
[MOCK - cursorTo(12, 5)]
[MOCK - cursorRestorePosition]



 pattern › f
[MOCK - cursorSavePosition]


 Pattern matches 2 tests from cached test suites

 › foo 1

 › foo 2
[MOCK - cursorTo(12, 5)]
[MOCK - cursorRestorePosition]



 pattern › f
[MOCK - cursorSavePosition]


 Pattern matches 2 tests from cached test suites

 › foo 1

 › foo 2
[MOCK - cursorTo(12, 5)]
[MOCK - cursorRestorePosition]
`;

exports[`shows the correct initial state 1`] = `
[MOCK - cursorHide]
[MOCK - clearScreen]

Pattern Mode Usage
 › Press Esc to exit pattern mode.
 › Press Enter to filter by a tests regex pattern.


[MOCK - cursorShow]



 pattern › 
[MOCK - cursorSavePosition]


 Start typing to filter by a test name regex pattern.
[MOCK - cursorTo(11, 5)]
[MOCK - cursorRestorePosition]
`;

exports[`shows the correct message when there are no cached tests 1`] = `
[MOCK - cursorHide]
[MOCK - clearScreen]

Pattern Mode Usage
 › Press Esc to exit pattern mode.
 › Press Enter to filter by a tests regex pattern.


[MOCK - cursorShow]



 pattern › 
[MOCK - cursorSavePosition]


 Start typing to filter by a test name regex pattern.
[MOCK - cursorTo(11, 5)]
[MOCK - cursorRestorePosition]
`;

exports[`test matching is case insensitive 1`] = `



 pattern › fO
[MOCK - cursorSavePosition]


 Pattern matches 2 tests from cached test suites

 › foo 1

 › foo 2
[MOCK - cursorTo(13, 5)]
[MOCK - cursorRestorePosition]
`;
